src/resolver.js

```diff
@@ -42,25 +39,36 @@ exports.resolve = (block, path, callback) => {
       // for the resolver
       node.links.forEach((l, i) => {
         const link = l.toJSON()
-        values[i] = link.multihash
+        values[i] = {
+          hash: link.multihash,
+          name: link.name,
+          size: link.size
```

Comment:
It is the size of this branch of the graph so far. It is needed to avoid traversing graphs over and over. go-ipfs has it since forever.